dandischema/models.py

```diff
@@ -510,7 +538,12 @@ def ensure_value(cls, val: Union[Any, List[Any]]) -> Union[Any, List[Any]]:
     _ldmeta = {"nskey": "schema"}
 
 
-PropertyValue.update_forward_refs()
+# This is mostly not needed at all since self-referencing models
```

Comment:
No, I see that the only forward reference in `PropertyValue` is refers to itself and according to [the documentation](https://docs.pydantic.dev/latest/concepts/postponed_annotations/#self-referencing-or-recursive-models), "Self-referencing fields will be automatically resolved after model creation".

I didn't remove it because it is mostly harmless for a complete model, calling `model_rebuild()` will not trigger another rebuild of the core schema, and I didn't want change much of the logic in this PR that's meant to update to Pydantic V2.

However, let me know if you refer to just remove the `model_rebuild()` calls in this PR.

Reply:
I will remove this in a subsequent PR since it is not really related to updating to Pydantic V2 (the behavior is the same in Pydantic V1).